### DISCUSSION TITLE
Next-auth is not fetching user with type-graphql request object

### DISCUSSION THREAD

**Discussion by vincent-thomas** [MichalLytek/type-graphql#1327] [Q&A]
**Describe the Bug**
The issue is that i can't get the user on server-side with next-auth and next.js. I do have access to the request object. (I guess type-graphql's `req` object follows the built-in 'Request' interface). I have tried with unstable_getServerSession, getSession and getToken from `next-auth`. None of which worked.

**To Reproduce**
Run
```
git clone https://github.com/vincent-thomas/typegraphql-with-next-auth-issue && \
yarn install && \ # Yarn is required because i'm using pnp
docker compose -d && yarn develop
// Not a pro on bash, you may have to modify it a bit
```

**Expected Behavior**
That it fetches the user as expected on the server with type-graphql 

**Logs**
```ts
// backend/resolvers/dog/resolver.ts
@Resolver(DogObject)
export class DogResolver {
  @Query(() => [DogObject])
  async dogs(@Ctx() { req }: IGraphqlContext): Promise<DogObject[]> {
    const user = await getToken({ req }); // <- 'req' is the request object
    console.log("USER", user); // <- 'user' = null, outputs "USER null"
    return dogs;
  }
```
**Environment (please complete the following information):**

- OS: Windows 11 (though running wsl 2 ubuntu with this issue)
- Node v16.16.0
- Package version 1.1.1
- TypeScript version 4.74

**Comment by MichalLytek** [comment]
TypeGraphQL `@Ctx()` inject the context you initalize in your sever handler. So please refer to the next docs about graphql endpoint.
